core/txpool/legacypool/legacypool.go

```diff
@@ -1723,7 +1725,9 @@ func (pool *LegacyPool) demoteUnexecutables() {
 			pool.all.Remove(hash)
 			log.Trace("Removed old pending transaction", "hash", hash)
 		}
-		balance := core.GetEffectiveGasBalance(pool.currentState, pool.chainconfig, addr)
+		balance, sgtBalance := core.GetGasBalances(pool.currentState, pool.chainconfig, addr)
```

Comment:
What is the reason that we want to relax the txpool filtering condition?

Reply:
In the case of `tx.value == acc.balance`, and `tx.value > sgtBalance`, passing either `acc.balance` or `acc.sgtBalance` as `balance` to filter the tx because of the condition `tx.Cost() <= balance`.  By relaxing `balance = acc.balance + acc.sgtBalance`, the tx will not be filtered out.

Note that the filter here can be relaxed: it is not guaranteed that all txs of the same account can be executed successfully.

Reply:
What's the worst case for this relax, will it lead to DOS ?

Reply:
The original filter is already relaxed, but definitively we can do more analysis here (or put TODO here)

Reply:
Originally if the relax happens, the balance will become zero quickly, so there's no much room for DOS. But now the evil user may deliberately construct transactions with value that can slip into the tx pool and potentially cause DOS.